packages/firebase_messaging/firebase_messaging_web/lib/firebase_messaging_web.dart

```diff
@@ -146,8 +141,12 @@ class FirebaseMessagingWeb extends FirebaseMessagingPlatform {
   }
 
   @override
-  Future<void> setForegroundNotificationPresentationOptions(
-      {required bool alert, required bool badge, required bool sound}) async {
+  Future<void> setForegroundNotificationPresentationOptions({
+    required bool alert,
+    required bool badge,
+    required bool sound,
+  }) async {
+    // TODO(rrousselGit) dead code? Should this throw an UnimplementedError?
```

Comment:
This is I guess a decision that needs making, web doesn't support this and therefore it was added as no-op function in case user doesn't conditionally call it, but there's also an argument to be had for the user calling it and expecting to do it something.

So it should probably throw a `UnimplementedError` as suggested, but this would be a breaking change and probably best not to have it as part of this PR?

Reply:
Agreed, hence the TODO 😄